File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerContract.java
package uk.co.sentinelweb.bitwatcher.common.ui.picker;

import android.support.annotation.NonNull;

import java.util.List;

public interface PickerContract {

    interface Presenter<T> {

        void bindData(@NonNull List<PickerItemModel<T>> items);

        void setInitialPosition(int position);

        void updatePositionFromScroll(int position);

        @NonNull
        PickerItemModel<T> getSelected();
    }

    interface View<T> {

        void setPresenter(@NonNull Presenter<T> presenter);

        void setData(@NonNull List<PickerItemModel<T>> items);

        void setInitialPosition(int position);
    }
}

File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerContract.kt
package uk.co.sentinelweb.bitwatcher.common.ui.picker

interface PickerContract {

    interface Presenter<T> {

        val selected: PickerItemModel<T>

        fun bindData(items: List<PickerItemModel<T>>)

        fun setInitialPosition(position: Int)

        fun updatePositionFromScroll(position: Int)
    }

    interface View<T> {

        fun setPresenter(presenter: Presenter<T>)

        fun setData(items: List<PickerItemModel<T>>)

        fun setInitialPosition(position: Int)
    }
}
